File: Ruby_Prueba/Practica4/lib/SuppliesPackage.rb
# encoding: utf-8

require_relative 'Dice'
module Deepspace

class SuppliesPackage

  def initialize(amP, flUn, shP)
    @ammoPower   = amP
    @fuelUnits   = flUn
    @shieldPower = shP
    dice = Dice.new

    if dice.receiveComodin
      @comodin = true
    else
      @comodin = false
    end
    
  end

  def self.newCopy(s)
    new(s.ammoPower,s.fuelUnits,s.shieldPower)
  end

  def ammoPower
    return @ammoPower
  end

  def fuelUnits
    return @fuelUnits
  end
  
  def shieldPower
    return @shieldPower
  end

  def comodin
    return @comodin
  end

  def to_s
    out = "AmmoPower: " + ammoPower().to_s + "\nFuelUnits: " + fuelUnits().to_s + "\nShieldPower: " + shieldPower().to_s 

  end

end #class

end #module

File: Java_Prueba/deepspace/SuppliesPackage.java
package deepspace;

public class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammoPower, float fuelUnits, float shieldPower){
        this.ammoPower = ammoPower;
        this.fuelUnits = fuelUnits;
        this.shieldPower = shieldPower;
    }

    SuppliesPackage(SuppliesPackage s){
        this (s.ammoPower,s.fuelUnits,s.shieldPower);
    }

    public float getAmmoPower(){ 
        return ammoPower;
    }
        

    public float getFuelUnits(){
        return fuelUnits;
    }

    public float getShieldPower(){
        return shieldPower;
    }
    
    public String toString() {
        String cadena = "AmmoPower: " + getAmmoPower() + "\nFuelUnits: " + getFuelUnits() +
        "\nShieldPower: " + getShieldPower();

        return cadena;
    }
}